opencog/atoms/core/Variables.cc

```diff
@@ -1317,12 +1337,48 @@ void Variables::extend(const Variables& vset)
 				_simple_typemap.insert({h, typemap_it->second});
 			}
 		}
+		// extend _glob_interval_map
+		extend_interval(h, vset);
 	}
 
 	// If either this or the other are ordered then the result is ordered
 	_ordered = _ordered or vset._ordered;
 }
 
+inline double max(double ld, double rd)
+{
+	if (ld < 0 and rd != std::numeric_limits<double>::infinity()) return ld;
+	if (rd < 0 and ld != std::numeric_limits<double>::infinity()) return rd;
```

Comment:
This seems .. weird... and hard to understand.  I'm not convinced this is correct...

Reply:
I think it's because negative value are +inf, thus greater than any other value. The unit tests are rather thorough, so I suppose it's correct.